### PR TITLE
Clean up the JS to be far more sensible

### PR DESCRIPTION
**Pull request by nickgartmann**
@peregrine @brookse @civilframe

Moved the game networking stuff off to its own factory object and wrapped it
events in $scope.$apply() so that we dont have to deal with that peppered 
throughout the game logic.

### CODE REVIEW

File: web/static/js/app.js [hunk 1]
```diff
@@ -1,125 +1,87 @@
...
-        $scope.$apply(function() {
-          if($scope.started) {
-            alert("Your opponent left.");
-            window.location.href=window.location.href;
-          } else {
-            $scope.opponents = [];
-          }
-        });
-      })
+  game.join().then(_ => {
```
Comment (civilframe) [inline web/static/js/app.js]: What's the _ for?

Comment (nickgartmann) [inline web/static/js/app.js]: in ES6, you can leave the parens off of fat arrow function definitions if you name a single parameter. The convention in elixir for when you don't care about the value of a parameter is `_` so i merged the two concepts

File: web/static/js/app.js [hunk 2]
```diff
@@ -1,125 +1,87 @@
...
+  });
+
+  game.on('fin', _ => {
+    $scope.complete = true;
+  });
+
+  // mutating the state directly allows
+  // us to not end up redrawing the
+  // board every time
+  game.on('click', clickedTile => {
```
Comment (civilframe) [inline web/static/js/app.js]: You might want to think about using angular events on $rootScope. Might be handy.

File: web/static/js/app.js (hunk 1 above)
Comment (civilframe) [inline web/static/js/app.js]: ooooooooh :+1:

**Comment by civilframe** [comment]
Nice work :+1:

**Pull request merged**